lib/puppet/provider/zabbix_proxy/ruby.rb

```diff
@@ -60,12 +61,46 @@ def create
   end
 
   def exists?
-    check_proxy(@resource[:hostname])
+    @property_hash[:ensure] == :present
   end
 
   def destroy
     zbx.proxies.delete([zbx.proxies.get_id(host: @resource[:hostname])].flatten)
   end
 
+  #
+  # zabbix_proxy properties
+  #
   mk_resource_methods
+
+  # We have to update all parameter of the interface and mode because
+  # it might not be there if we switch the proxy from active to passive.
+  def update_mode_interface
+    zbx.proxies.update(
+      proxyid: @property_hash[:id],
+      status: @resource[:mode] + 5, # Normally 0 is active and 1 is passive, in the API, its 5 and 6
+      interface: {
+        ip: @resource[:ipaddress],
+        dns: @resource[:hostname],
+        useip: @resource[:use_ip] ? 1 : 0,
+        port: @resource[:port]
```

Comment:
What happens if eg `@resource[:port]` is `undef`?  Does it need to be `port: @resource[:port] || @property_hash[:port]`